atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/pue/ResilientCommitTimestampPutUnlessExistsTable.java

```diff
@@ -84,7 +94,7 @@ private Map<Long, Long> processReads(Map<Cell, byte[]> reads, Map<Long, Cell> st
             PutUnlessExistsValue<Long> currentValue = encodingStrategy.decodeValueAsCommitTimestamp(startTs, actual);
 
             Long commitTs = currentValue.value();
-            if (currentValue.isCommitted()) {
+            if (currentValue.isCommitted() || acceptStagingReadsOnVersionThree.get()) {
```

Comment:
I might have missed something but should this not be something like

```suggestion
            if (currentValue.isCommitted() || (acceptStagingReadsOnVersionThree.get() && currentValue.isStaging())) {
```

Reply:
A fair question - though yeah a value is either committed or staging, so if we got to that branch it would definitively be staging.